lib/tasks/community_api_import.rake

```diff
@@ -4,11 +4,31 @@ namespace :community_api do
   desc 'Import data from Community API'
   task import: :environment do |_task|
     Rails.logger = Logger.new(STDOUT)
+    ActiveRecord::Base.connection.disable_query_cache!
 
-    Prison.active.each do |prison|
-      prison.offenders.each do |offender|
+    # Potential memory leak:
+    # prison.offenders seems to be cached somewhere, because subsequent calls to prison.offenders are always faster than the first call
```

Comment:
This is our cache in redis - so IMHO there isn't a memory leak here. There's also a good chance that we hit the Oracle cache on the far end (Oracle caches all queries based on parameters when it can)

Reply:
Good point @starswan – thanks 👍🏻

I've just confirmed this with:

```ruby
Prison.new('LEI').count # first call is slow
Prison.new('LEI').count # second call is quick
Rails.cache.clear
Prison.new('LEI').count # is slow again after clearing the cache
```

I'll remove this comment.